store/sqlstore/group_supplier.go

```diff
@@ -765,3 +765,92 @@ func groupSyncableToGroupChannel(groupSyncable *model.GroupSyncable) *groupChann
 		ChannelId:     groupSyncable.SyncableId,
 	}
 }
+
+// TeamMembersToRemove returns all team members that should be removed based on group constraints.
+func (s *SqlSupplier) TeamMembersToRemove(ctx context.Context, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	sql := `
+		SELECT
+			TeamMembers.*
+		FROM
+			TeamMembers
+			JOIN Teams ON Teams.Id = TeamMembers.TeamId
+		WHERE
+			TeamMembers.DeleteAt = 0
+			AND Teams.DeleteAt = 0
+			AND Teams.GroupConstrained = TRUE
+			AND (TeamMembers.TeamId, TeamMembers.UserId)
+			NOT IN (
+				SELECT
+					Teams.Id AS TeamId, GroupMembers.UserId
+				FROM
+					Teams
+					JOIN GroupTeams ON GroupTeams.TeamId = Teams.Id
+					JOIN UserGroups ON UserGroups.Id = GroupTeams.GroupId
+					JOIN GroupMembers ON GroupMembers.GroupId = UserGroups.Id
+				WHERE
+					Teams.GroupConstrained = TRUE
+					AND GroupTeams.DeleteAt = 0
+					AND UserGroups.DeleteAt = 0
+					AND Teams.DeleteAt = 0
+					AND GroupMembers.DeleteAt = 0
+				GROUP BY
+					Teams.Id,
+					GroupMembers.UserId)`
```

Comment:
This query (and the channel one) are the meat of this PR. 

It's getting all of the **allowed** team members (based on `GroupTeams` associations) for all of the group-constrained teams, for example:

|`TeamId`|`UserId`|
|-|-|
|`t1`|`u1`|
|`t1`|`u2`|
|`t2`|`u1`|

and comparing against the **actual** team members, for example:

|`TeamId`|`UserId`|
|-|-|
|`t1`|`u1`|
|`t1`|`u3`|

and then returning the users that are **actual** members but `NOT IN` **allowed** members:

|`TeamId`|`UserId`|
|-|-|
|`t1`|`u3`|

and that is the list of users that are "ToRemove".